base/string_utils.hpp

```diff
@@ -306,6 +306,11 @@ void Tokenize(string const & str, char const * delims, TFunctor && f)
   }
 }
 
+/// Splits a string by the delimiter, keeps empty parts, on an empty string returns an empty vector.
+/// Does not support quoted columns, newlines in columns and escaped quotes.
+/// @return false if the line is empty or number of columns differs from |columns|.
+bool ParseCSVRow(string const & s, vector<string> & target, char const delimiter = ',', size_t const columns = 0);
```

Comment:
Agree, but I'm against the possible value of zero to columns. CSV with zero number of columns is invalid, therefore this function shouldn't be called with columns parameter equal to zero. As it's quite hard with C++ type system to express values that greater than zero, I'm against the columns parameter at all.